Annotato/Annotato/Views/Document/Edit/DocumentView.swift

```diff
@@ -16,16 +18,105 @@ class DocumentView: UIView {
         self.layer.borderColor = UIColor.black.cgColor
 
         addTapGestureRecognizer()
-        initializeAnnotationViews()
+        addObservers()
+        initializePdfView()
+        initializeAnnotationViewsForVisiblePages()
     }
 
-    private func initializeAnnotationViews() {
+    private func addObservers() {
+        NotificationCenter.default.addObserver(
+            self, selector: #selector(handleVisiblePageChange(notification:)),
+            name: Notification.Name.PDFViewVisiblePagesChanged, object: nil
+        )
+    }
+
+    @objc
+    private func handleVisiblePageChange(notification: Notification) {
+        initializeAnnotationViewsForVisiblePages()
+    }
+
+    private func checkIfAnnotationIsInVisiblePages(
+        anno: DocumentAnnotationViewModel,
+        visiblePages: [PDFPage]
+    ) -> Bool {
+        guard let pageNumForAnnotation = anno.associatedPage.label else {
+            return false
+        }
+        let visiblePagesIndex = visiblePages.map({ pdfPage -> String in
+            guard let label = pdfPage.label else {
+                return "-1"
+            }
+            return label
+        })
+        return visiblePagesIndex.contains(pageNumForAnnotation)
+    }
+
+    private func checkIfAnnoIsAlreadyInSubview(
```

Comment:
Shall we not use abbreviations? It would be more consistent if either everything uses a standard abbreviation or we write in full. Also check seems unnecessary as `Bool` functions just need to sound like assertions.
```suggestion
    private func annotationIsAlreadyInSubview(
```

Reply:
Ok done